### ISSUE TITLE
Editing the task code for a check creates inconsistent UI experience

### ISSUE THREAD

**Issue by aanthony1243** [influxdata/ui#1783]
Thank you for reporting a bug in InfluxData UI.

- Please ask usage questions on the Influx Community site.
  - https://community.influxdata.com/
- Please add a :+1: or comment on a similar existing bug report instead of opening a new one.
  - https://github.com/influxdata/ui/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+is%3Aclosed+sort%3Aupdated-desc+label%3Akind%2Fbug+
- Please check whether the bug can be reproduced with the latest release.
- The fastest way to fix a bug is to open a Pull Request.
  - https://github.com/influxdata/ui/pulls

**Steps to reproduce:**
List the minimal actions needed to reproduce the behavior.

1. Create a task via the UI
2. Save it
3. In the configuration options for the check, choose "edit task" 

it shows this message: 
![Screen Shot 2021-06-23 at 10 47 06 AM](https://user-images.githubusercontent.com/25036188/123118134-5d036180-d410-11eb-9fa7-741e457ae3a1.png)

but this is not true.  I can delete all the code and write something new, and save it.   Then I can go back to the check and the editor opens with all the previous values.  If I save it, it overwrites the custom code I wrote.  

**Expected behavior:**
if I edit the code directly, it should be converted to a regular task that is no longer editable by the C+N interface.  This task should be recognized as no longer being an alert and be displayed in the tasks list page.  

**Actual behavior:**
I can edit the code and then without any warning, that code can be overwritten. 

** Side Effects ** 
Any action that regenerates checks or has some assumption that installed checks are all on the same template will have inconsistent results with a check that is manually edited.  

**Suggestions to fix:**
allowing the user to see the code read-only would be safe.  

`convert to task` would be acceptable too, if they require editing the code directly.  

the checks UI needs to inspect the task code and make sure that it matches the expected template before it allows editing/saving it using the UI dialog.   

**Note:**
This bug identically impacts notification rules.  

**Environment info:**
all cloud environments.

**Comment by russorat** [comment]
@aanthony1243 let's explore the "convert to task" option here. Just thinking about this in my head, could we just remove the check/notification resource without deleting the underlying task? but then we'd need a way to convert the underlying task from an internal one to a user facing one so that it would be returned in the GET /tasks call.

**Comment by aanthony1243** [comment]
that's where my mind is drifting too.  I'll talk to my team about it and report back.

**Comment by aanthony1243** [comment]
@russorat   this would also have implications on task limits for free tier users.

**Comment by brettbuddin** [comment]
Agreed, it would need to be transformed into a standard `system` Task. What I think we should do there is copy the existing underlying Task record (creating a new record) and then let the deletion of the Check or Notification Rule delete the Task it owns without having to do any changes to that logic. That way it's all additive in code and we won't have to be messing around in the internals of the existing C+N system on the platform side.

**Comment by russorat** [comment]
@brettbuddin that certainly sounds clean. the only downside i can think of is the user loses their run history. i'm honestly not sure if that's a huge issue or not.

**Comment by aanthony1243** [comment]
the run history would still be present in the _monitoring table.  And that history only lasts 3 days so by the end of a week they'd be no different than before.

**Comment by danathanL** [comment]
I think this is actually very similar, if not the same issue as I am currently facing with alerts...
